### PR TITLE
Implement Export Block Proposals

### PR DESCRIPTION
**Pull request by rauljordan**
**What type of PR is this?**

> Feature

**What does this PR do? Why is it needed?**

This PR implements the export block proposal history according to https://eips.ethereum.org/EIPS/eip-3076.

**Which issues(s) does this PR fix?**

Part of #7813

### CODE REVIEW

File: validator/slashing-protection/local/standard-protection-format/export.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+		}
+		dataByPubKey[pubKey] = &ProtectionData{
+			Pubkey:             pubKeyHex,
+			SignedBlocks:       signedBlocks,
+			SignedAttestations: nil,
+		}
+	}
+
+	// Next we turn our map into a slice as expected by the EIP-3076 JSON standard.
+	dataList := make([]*ProtectionData, 0)
```
Comment (prestonvanloon) [inline validator/slashing-protection/local/standard-protection-format/export.go:50]:
```suggestion
	dataList := make([]*ProtectionData, 0, len(dataByPubKey)
```

File: validator/slashing-protection/local/standard-protection-format/export.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	}
+	interchangeJSON.Metadata.GenesisValidatorsRoot = genesisRootHex
+	interchangeJSON.Metadata.InterchangeFormatVersion = INTERCHANGE_FORMAT_VERSION
+
+	// Extract the existing public keys in our database.
+	proposedPublicKeys, err := validatorDB.ProposedPublicKeys(ctx)
+	if err != nil {
+		return nil, err
+	}
+	dataByPubKey := make(map[[48]byte]*ProtectionData)
```
Comment (prestonvanloon) [inline validator/slashing-protection/local/standard-protection-format/export.go:30]:
```suggestion
	dataByPubKey := make(map[[48]byte]*ProtectionData, len(proposedPublicKeys))
```

File: validator/slashing-protection/local/standard-protection-format/import.go [hunk 3]
```diff
@@ -74,7 +75,14 @@ func ImportStandardProtectionJSON(ctx context.Context, validatorDB db.Database,
 	// until after we successfully parse all data from the JSON file. If there is any error
 	// in parsing the JSON proposal and attesting histories, we will not reach this point.
 	for pubKey, proposalHistory := range proposalHistoryByPubKey {
+		bar := initializeProgressBar(
+			len(proposalHistory.Proposals),
+			fmt.Sprintf("Importing past proposals for validator public key %#x", bytesutil.Trunc(pubKey[:])),
+		)
 		for _, proposal := range proposalHistory.Proposals {
+			if err := bar.Add(1); err != nil {
+				log.Debug("Could not increase progress bar")
```
Comment (prestonvanloon) [inline validator/slashing-protection/local/standard-protection-format/import.go:84]:
```suggestion
				log.WithError(err).Debug("Could not increase progress bar")
```

File: validator/slashing-protection/local/standard-protection-format/export.go (hunk 2 above)
Comment (rauljordan) [reply validator/slashing-protection/local/standard-protection-format/export.go:30]: It actually also depends on attested public keys, so will do in a separate PR

**Review by prestonvanloon — DISMISSED** [review]

**Review by prestonvanloon — APPROVED** [review]

File: validator/slashing-protection/local/standard-protection-format/export.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+		dataByPubKey[pubKey] = &ProtectionData{
+			Pubkey:             pubKeyHex,
+			SignedBlocks:       signedBlocks,
+			SignedAttestations: nil,
+		}
+	}
+
+	// Next we turn our map into a slice as expected by the EIP-3076 JSON standard.
+	dataList := make([]*ProtectionData, 0)
+	for _, item := range dataByPubKey {
```
Comment (nisdas) [inline validator/slashing-protection/local/standard-protection-format/export.go:51]: This loop wasn't needed, it could have been done in the first loop.

**Pull request merged**